flang/lib/Lower/OpenMP/OpenMP.cpp

```diff
@@ -498,6 +502,8 @@ struct OpWithBodyGenInfo {
   mlir::Location loc;
   /// [in] current PFT node/evaluation.
   Fortran::lower::pft::Evaluation &eval;
+  /// [in] kind of operation.
+  OpKind kind;
```

Comment:
Consider adding more documentation on what the different values mean.

Why introduce a new enum and not reusing `llvm::omp::Directive`?

Reply:
I guess the solution should be the same as in the other PR.

Reply:
Note that LLVMFrontend defines helpers such as `isOpenMPParallelDirective` and `getDirectiveAssociation` to find out whether an `llvm::omp::Directives` involves a loop/parallel.

Reply:
> Why introduce a new enum and not reusing `llvm::omp::Directive`?

I agree there's not much point to adding that enum. I changed it to `llvm::omp::Directive` now.

> Note that LLVMFrontend defines helpers such as `isOpenMPParallelDirective` and `getDirectiveAssociation` to find out whether an `llvm::omp::Directives` involves a loop/parallel.

Thanks for the pointer to `getDirectiveAssociation`. I updated `createBodyOfOp` to use it. We don't need to use `isOpenMPParallelDirective` in this case, since the directive here should refer to a leaf construct, so we don't need to deal with compound ones.